src/ln/channel.rs

```diff
@@ -1092,28 +1167,50 @@ impl Channel {
 	// Message handlers:
 
 	pub fn accept_channel(&mut self, msg: &msgs::AcceptChannel) -> Result<(), HandleError> {
+		macro_rules! send_error_message {
+			( $msg: expr ) => {
+				return Err(HandleError{err: $msg, action: Some(msgs::ErrorAction::SendErrorMessage{ msg: msgs::ErrorMessage { channel_id: msg.temporary_channel_id, data: $msg.to_string() }})});
+			}
+		}
 		// Check sanity of message fields:
 		if !self.channel_outbound {
-			return Err(HandleError{err: "Got an accept_channel message from an inbound peer", action: None});
+			send_error_message!("Got an accept_channel message from an inbound peer");
 		}
 		if self.channel_state != ChannelState::OurInitSent as u32 {
-			return Err(HandleError{err: "Got an accept_channel message at a strange time", action: None});
+			send_error_message!("Got an accept_channel message at a strange time");
 		}
 		if msg.dust_limit_satoshis > 21000000 * 100000000 {
-			return Err(HandleError{err: "Peer never wants payout outputs?", action: None});
+			send_error_message!("Peer never wants payout outputs?");
 		}
 		if msg.channel_reserve_satoshis > self.channel_value_satoshis {
-			return Err(HandleError{err: "Bogus channel_reserve_satoshis", action: None});
+			send_error_message!("Bogus channel_reserve_satoshis");
+		}
+		if msg.dust_limit_satoshis > msg.channel_reserve_satoshis {
+			send_error_message!("Bogus channel_reserve and dust_limit");
+		}
+		if msg.channel_reserve_satoshis < self.our_dust_limit_satoshis {
+			send_error_message!("Peer never wants payout outputs?");
+		}
+		if msg.dust_limit_satoshis > self.our_channel_reserve_satoshis {
+			send_error_message!("Dust limit is bigger than our channel reverse");
+		}
+		if msg.channel_reserve_satoshis < self.our_dust_limit_satoshis {
```

Comment:
This check is the same as the one two up?

Reply:
just double checking :)